pkg/sitewise/client/client.go

```diff
@@ -289,17 +299,12 @@ func GetClient(region string, settings models.AWSSiteWiseDataSourceSetting, prov
 				},
 			},
 		}
-		httpClient := &http.Client{Transport: tr}
 
-		swcfg = swcfg.WithHTTPClient(httpClient)
-		swcfg = swcfg.WithDisableEndpointHostPrefix(true)
+		options = append(options, awsconfig.WithHTTPClient(&http.Client{Transport: tr}))
+		// TODO: figure out how to replace this with smithy's version
+		// https://pkg.go.dev/github.com/aws/smithy-go/transport/http#DisableEndpointHostPrefix
+		//swcfg = swcfg.WithDisableEndpointHostPrefix(true)
```

Comment:
Would this work?

```go
import (
  // ... other imports
  smithyhttp "github.com/aws/smithy-go/transport/http"
)
		// ... rest of code
		options = append(options, awsconfig.WithHTTPClient(&http.Client{Transport: tr}))
		ctx = smithyhttp.DisableEndpointHostPrefix(ctx, true)
```

For this Edge region path I'm not sure if we have something set up for testing it. We'll either have to set it up or ask someone on the SiteWise team to verify it.